ci-operator/step-registry/cucushift/installer/check/vsphere/zone/cucushift-installer-check-vsphere-zone-commands.sh

```diff
@@ -0,0 +1,161 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+set -x
+export KUBECONFIG=${SHARED_DIR}/kubeconfig
+source "${SHARED_DIR}/govc.sh"
+source "${SHARED_DIR}/vsphere_context.sh"
+INSTALL_CONFIG="${SHARED_DIR}/install-config.yaml"
+
+readarray -t zones_region_from_config < <(yq-go r ${INSTALL_CONFIG} 'platform.vsphere.failureDomains[*].region')
+readarray -t zones_zone_from_config < <(yq-go r ${INSTALL_CONFIG} 'platform.vsphere.failureDomains[*].zone')
+for ((i=0;i<${#zones_zone_from_config[@]};i++)); do
+    if [[ $(yq-go r ${INSTALL_CONFIG} 'platform.vsphere.failureDomains[*].topology.datacenter' | wc -l) == "0" ]];then
+        readarray -t zones_datacenter_from_config < <(yq-go r ${INSTALL_CONFIG} 'platform.vsphere.datacenter')
+    else
+	readarray -t zones_datacenter_from_config < <(yq-go r ${INSTALL_CONFIG} 'platform.vsphere.failureDomains[*].topology.datacenter')
+    fi
+done
+infra_id=$(oc get -o jsonpath='{.status.infrastructureName}{"\n"}' infrastructure cluster)
+
+function check_rhcos() {
+    check_result=0
+    # shellcheck disable=SC2207
+    vsphere_datacenters=($(govc ls /))
```

Comment:
folder creation is based on datacenter defined under platform.vsphere or platform.vsphere.failureDomain in install-config. I think you need to read datacenters from install-config, then check folder creation under each datacenter.